definitions/npm/zen-observable_v0.8.x/flow_v0.104.x-/zen-observable_v0.8.x.js

```diff
@@ -0,0 +1,38 @@
+declare module 'zen-observable' {
+  declare export interface SubscriptionObserver<T> {
+    next(value: T): void;
+    error(err: Error): void;
+    complete(): void;
+    get closed(): boolean;
+  }
+
+  declare export interface Observer<T> {
+    next?: (value: T) => any;
+    error?: (err: Error) => any;
+    complete?: () => any;
```

Comment:
Possibly use `mixed`?

Reply:
That works for arguments that are functions, but apparently the variance doesn't work for methods of an object argument:

```
    
    Error ------------------------------------------------------------------------------------- test_zen-observable.js:35:26
    
    Cannot call `observable.subscribe` because undefined [1] is incompatible with mixed [2] in property `complete`.
    
       test_zen-observable.js:35:26
                                    v---------------------
       35|     const subscription = observable.subscribe({
       36|       complete() {},
       37|     });
               -^
    
    References:
       test_zen-observable.js:36:17
       36|       complete() {},
                           ^ [1]
       zen-observable_v0.8.x.js:12:22
       12|     complete?: () => mixed;
                                ^^^^^ [2]
    
    
    
    Found 2 errors
```